### DISCUSSION TITLE
Volume inconsistent in Hydrogen Synth

### DISCUSSION THREAD

**Discussion by Djard** [hydrogen-music/hydrogen#1951] [Hydrogen on Windows]
Has anybody experienced a drop in playback volume of the Hi-Hat Closed when the Kick Drum is added. In Pattern 1, I wrote a rhythm for the Hi-Hat (1/8th notes). In Pattern 2, I added a Kick drum (on 1st and 2nd beat). The volume of the Hi-Hat drops about 6 dB when the Kick drum is added. I can correct the issue by increasing the velocity of the H.H. in Pattern 2, but that should not be necessary. If someone has encountered this problem and found a workaround, please share.

**Comment by theGreatWhiteShark** [comment]
Hmm.. this sounds a bit like a compressor issue further down the signal chain. Do you see the volume drop in Hydrogen's mixer as well? Could you share the song?

**Reply by Djard** [reply]
Thanks for responding. Please find a copy of the song with the issue I described that is isolated to the first pattern. To compensate for the inconsistent playback volume, I lowered the velocity of the Hi-Hat Closed in the "Intro 1" pattern (first 2 bars). I use a custom kit with the Linn Drum kit that I customized. I have used this kit many times, without any problems, so my guess is that the new version of Hydrogen does not like the kit.

I did try creating a new pattern and deleting the one with the problem, but the playback volume remained louder than ones in subsequent patterns.

**Reply by Djard** [reply]
I was unable to add the file with *.h2song file extension, so I tried adding it as a zip file; but I keep getting notice that "This file is hidden. Try again with another file." Sorry. If there is another way to add the file, you will need to let me know, and I will try again.

Uh-oh, the browse option might work....

[.A Good Old Tyme Melody.zip](https://github.com/hydrogen-music/hydrogen/files/14625411/default.A.Good.Old.Tyme.Melody.zip)

**Reply by theGreatWhiteShark** [reply]
Thanks.

Hmm... Maybe it is related to your drumkit. Could you export it and upload the resulting .h2drumkit file? I can not reproduce the issue with our stock kits.

In addition, could you set "Meters falloff speed" in Preferences > Appearance > Interface to `Fast` and check for the peak values of "Hi-Hat Closed" displayed above the volume slider in the mixer? Does the numerical value drop as well?

**Comment by Djard** [comment]
Attached is the drumkit file, also a clip of what I experience, wherein the volume of the HH drops after the first two bars.

I doubt the issue is with the kit as it worked properly in the earlier version of Hydrogen. But maybe something changed on my system.
I set the "Meters falloff speed" to Fast and played the song. None of the numbers in the Mixer changed; the LEDs of each instrument flash with equal strength. 
Hope you can figure out the problem.
Kind regards,Gerard


    On Monday, March 18, 2024 at 03:06:40 PM EDT, theGreatWhiteShark ***@***.***> wrote:  
 
 


Thanks.

Hmm... Maybe it is related to your drumkit. Could you export it and upload the resulting .h2drumkit file? I can not reproduce the issue with our stock kits.

In addition, could you set "Meters falloff speed" in Preferences > Appearance > Interface to Fast and check for the peak values of "Hi-Hat Closed" displayed above the volume slider in the mixer? Does the numerical value drop as well?

**Reply by theGreatWhiteShark** [reply]
> None of the numbers in the Mixer changed;

That indicates that all notes are played back with using the same volume.

Do earlier versions of Hydrogen still work without those volume drops? Maybe it is your system after all. If so, the drop should not be present when exporting the song into an audio file. Even with the most recent version of Hydrogen. Could you check?

**Comment by cme** [comment]
Hi @Djard , looks like the drumkit and clip didn't attach :/ 

Another question, is this reproducible in pattern mode with each pattern, or just in song mode when it progresses from pattern 1 on to pattern 2?

Thanks!

**Comment by Djard** [comment]
As an experiment, I wrote four measures of Hi-Hat notes in 4/4. (pattern 1) In the 5th measure (pattern 2), I added a Kick drum on the third beat. Then I wrote 4 more measures of HH (pattern 3). The HH volume did not drop in the 5th measure, until the kick drum played. The volume of the HH gradually got louder: by the 10th measure, the playback volume of the HH was gradually restored, like a slow crescendo). I then replaced the Kick drum with a snare, just to see if the Kick drum caused the instability. The problem was replicated. So adding instruments causes an automatic volume change as if hydrogen wants to even the volume.
I don't have a copy of the earlier version. The one I am using is 1.2.0-'beta6' [Mar 31 2023]. I will update to the latest version and let you know if the problem persists.
Cheers!
Gerard 


 
    On Friday, March 22, 2024 at 07:19:41 AM EDT, theGreatWhiteShark ***@***.***> wrote:  
 
 



None of the numbers in the Mixer changed;


That indicates that all notes are played back with using the same volume.

Do earlier versions of Hydrogen still work without those volume drops? Maybe it is your system after all. If so, the drop should not be present when exporting the song into an audio file. Even with the most recent version of Hydrogen. Could you check?

**Reply by cme** [reply]
Excellent experiment. Was this using the standard Hydrogen drum kit?

There's nothing within Hydrogen itself that can cause this sort of behavior, so I concur with @theGreatWhiteShark that it sounds exactly there's a compressor somewhere downstream of Hydrogen itself.

Some things to try would be to change the PortAudio host API used in the audio settings dialog to see if there's one that removes the problem. Failing that, an interesting diagnostic test would be to

1. export audio as a WAV from Hydrogen
2. Play it back in Windows Media Player or similar and see if it has the problem
3. If you hear the same behaviour:
    - Check the WAV with Audacity or similar to make sure the hi-hat peaks are actually consistent.
       + If they're not, then that's actually a Hydrogen problem, which would be very odd.
       + If they're consistent then the problem is somewhere in Windows audio, drivers, OEM software etc.
4. If it sounds fine in Media Player etc:
    - load the WAV file into Hydrogen as a "Playback Track"
       + remove notes from patterns but add enough empty patterns to the song that Hydrogen will play the WAV file through
       + If it sounds compressed, then there's something specific about the connection between Hydrogen and the rest of the audio system.
        + if it sounds fine, then there's something very weird going on, possibly in Hydrogen.

I guess one thing to note is that if there's a compressor somewhere, the actual behaviour might be sensitive to gain level -- like if you have Hydrogen turned up in the Windows Mixer.

What does the rest of your audio system look like? Any third party software installed, laptop internal speakers, headphones etc?

It's an interesting problem anyway! Cheers.

**Reply by theGreatWhiteShark** [reply]
> If they're consistent then the problem is somewhere in Windows audio, drivers, OEM software etc.

This could e.g. happen with a firmware update of one of your audio devices. E.g. I recently had problems with my headset at work that almost drove me mad. After exactly one minute volume dropped by about 50% only to be at full level again at the start of the next song. First, I blamed Windows. But it happened on Linux as well. In the end it turned out to be the headset itself and I could fix it by not plugging it directly into the USB port of my laptop but into an USB hub instead. Crazy stuff you would never expect at first.

**Reply by Djard** [reply]
I am not using a compressor. I always export the file in WAV format. The sudden reduction in playback volume after an instrument is added (such as a kick drum or Tom) that recovers after a few beats is audible in Media Player Classic and Audacity. It is also audible in Hydrogen before exporting. I mostly use a customized Linn drum kit but the issue is present when using the system GMRockKit. I hear the drop in volume through my Sennheisser headphones as well as through my Yamaha 5.1 amp and Bose 802 speakers, including phone. When I create a drum track with other music software, like Ableton and Finale, no such problem occurs (they have their own issues).

Wait! I just tried cme's suggestion to change the Host API. Selecting the "Windows WDM-KS" portAudio driver lowered the volume a little but resolved the problem. Yay! Thank you soooooo much.

**Comment by Djard** [comment]
Thank you kindly for the info.


Yahoo Mail: Search, Organize, Conquer 
 
  On Mon, Mar 25, 2024 at 4:58 PM, ***@***.***> wrote:   



If they're consistent then the problem is somewhere in Windows audio, drivers, OEM software etc.


This could e.g. happen with a firmware update of one of your audio devices. E.g. I recently had problems with my headset at work that almost drove me mad. After exactly one minute volume dropped by about 50% only to be at full level again at the start of the next song. First, I blamed Windows. But it happened on Linux as well. In the end it turned out to be the headset itself and I could fix it by not plugging it directly into the USB port of my laptop but into an USB hub instead. Crazy stuff you would never expect at first.

**Comment by Djard** [comment]
I figured out the cause of the inconsistent audio volume that affected all playback apps. A Windows update enabled "Loudness Equalization." Disabling this enhancement in Windows' Sound properties (Control Panel) resolved the problem.
